api/src/pcapi/routes/adage_iframe/educational_institution.py

```diff
@@ -27,9 +28,17 @@ def get_educational_institution_with_budget(
     )
     institution = institution.one_or_none()
 
-    for deposit in institution.deposits:
-        if datetime.utcnow().year == deposit.educationalYear.beginningDate.year:
-            amount = deposit.get_amount()
+    try:
+        amount = [
+            deposit.get_amount()
+            for deposit in institution.deposits
+            if datetime.utcnow().year == deposit.educationalYear.beginningDate.year
+        ][0]
+    except IndexError:
+        raise ApiErrors(
+            {"global": "La récupération des informations du de l'institution avec budget via Adage a échoué"},
+            status_code=404,
+        )
```

Comment:
```suggestion
    for deposit in institution.deposits:
        if datetime.utcnow().year == deposit.educationalYear.beginningDate.year:
            amount = deposit.get_amount()
            break
    else:
        raise ApiErrors(
            {"global": "L'établissement scolaire ne semble pas avoir de budget pour cette année."},
            status_code=404,
        )
```

attention le `else` est aligné avec le `for` (il n'est executé que si le for arrive au bout sans jamais passer sur le break)